src/charm.py

```diff
@@ -211,8 +214,12 @@ def _on_pgbouncer_pebble_ready(self, event: PebbleReadyEvent) -> None:
         container.add_layer(PGB, pebble_layer, combine=True)
         container.autostart()
 
-        if self.check_pgb_running():
-            self.unit.status = ActiveStatus()
+        try:
+            if self.check_pgb_running():
+                self.unit.status = ActiveStatus()
+        except ConnectionError:
+            self.unit.status = WaitingStatus("pgbouncer not running")
+            event.defer()
```

Comment:
[Minor] I would prepend it with Debug message, to read `juju debug-log` with a pleasure.